src/averageAges.js

```diff
@@ -15,11 +15,19 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const mans = people.filter(({ sex }) => (sex === 'm'));
+
+  const sameCentury = mans.filter(({ died }) => (
+    Math.ceil(died / 100) === century
+  ));
+
+  const isCentury = (typeof century) === 'number' ? sameCentury : mans;
```

Comment:
```suggestion
```
Instead of creating so many variables you can filter `people` array in one place like
```
const men = people.filter(person =>
    century
      ? Math.ceil(person.died / 100) === century && person.sex === 'm'
      : (person.sex === 'm')
```

Here you just check if `century` is undefined or not.